### PR TITLE
Retain multiple branches for git repository

### PR DESCRIPTION
**Pull request by karthikeyanBD**

### CODE REVIEW

**Comment by Priya5** [comment]
LGTM

**Comment by notalex** [comment]
Please wait for a day before merging this. This change needs contemplation.

File: lib/ohloh_scm/git/scm.rb [hunk 1]
```diff
@@ -43,8 +43,10 @@ def fetch_new_commits(remote_scm)
       end
 
       def clone_and_create_tracking_branch(remote_scm)
-        run "rm -rf '#{url}'"
-        run "git clone -q -n '#{remote_scm.url}' '#{url}'"
+        unless status.scm_dir_exist?
+          run "rm -rf '#{url}'"
+          run "git clone -q -n '#{remote_scm.url}' '#{url}'"
```
Comment (notalex) [inline lib/ohloh_scm/git/scm.rb]: The **scm_dir** will always be a new directory when doing a refetch. So checking for its existence would not solve the issue.

Comment (santhanakarthikeyan) [reply lib/ohloh_scm/git/scm.rb]: The issue is not about checking the directory existence. When we are fetching the repository, In https://github.com/blackducksoftware/ohloh_scm/blob/master/lib/ohloh_scm/git/scm.rb#L30 we check if repo_dir has that branch. If not, we are removing the repo_dir and re-cloning entire repository. 

This fix will avoid refetching entire repository for each code_locations

Comment (notalex) [reply lib/ohloh_scm/git/scm.rb]: Lets discuss this on slack. I need some clarification.

**Pull request merged**